### PR TITLE
HAL_ChibiOS: improve throughput of slcan router

### PR DESCRIPTION
**Pull request by bugobliterator**

### CODE REVIEW

File: libraries/AP_HAL_ChibiOS/CANSerialRouter.cpp [hunk 1]
```diff
@@ -117,12 +117,14 @@ void SLCANRouter::slcan2can_router_trampoline(void)
         }
         chSysUnlock();
         _slcan_if.reader();
-        if (_can_tx_queue.available() && _can_if) {
+        while (_can_tx_queue.available() && _can_if) {
             _can_tx_queue.peek(it);
             if (_can_if->send(it.frame, uavcan::MonotonicTime::fromUSec(AP_HAL::micros64() + 1000), 0)) {
```
Comment (peterbarker) [inline libraries/AP_HAL_ChibiOS/CANSerialRouter.cpp:122]: Should these loops break if the send fails?

Comment (bugobliterator) [reply libraries/AP_HAL_ChibiOS/CANSerialRouter.cpp:122]: Yeah, It's probably better for them to just break on fail. I will add that.

Comment (bugobliterator) [reply libraries/AP_HAL_ChibiOS/CANSerialRouter.cpp:122]: This is resolved now.

**Comment by peterbarker** [comment]
@davidbuzz it it easy for you to verify this works to resolve the throughput problem you were having?

**Comment by davidbuzz** [comment]
no, not easy.

**Pull request merged**